### PR TITLE
Fixing serializer for objects as properties

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

- Fixing serializer for objects as properties.
- Migrating projects to support VS2022.

## Motivation and Context
- Fixing serializer for objects as properties
- Typically when you have a Hashtable and place a custom class in it

## How Has This Been Tested?

Unit test, added

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by josesimoes — CHANGES_REQUESTED** [review]
You seem to be using tabs for indentation now... We're using 4 spaces everywhere.

**Comment by Ellerbach** [comment]
> You seem to be using tabs for indentation now... We're using 4 spaces everywhere.

I just did ctrl+k+d to fix all the indentations issues in the test file.

File: nanoFramework.Json/JsonObject.cs [hunk 1]
```diff
@@ -246,7 +246,7 @@ public static JsonArray Serialize(ArrayList source)
 						result[index] = Serialize(valueType, item);
 					}
 				}
-
+
```
Comment (josesimoes) [inline nanoFramework.Json/JsonObject.cs:249]: Not on the test file. Here a line that should be just a line.

Comment (Ellerbach) [reply nanoFramework.Json/JsonObject.cs:249]: almost all files are tabs! I can fix that in another PR

**Review by josesimoes — APPROVED** [review]
Let's have the **real** fix on this PR.
And the formatting issues on a subsequent one.

**Pull request merged**